rasa/nlu/components.py

```diff
@@ -87,16 +87,41 @@ def validate_required_components_from_data(
     pipeline: List["Component"], data: TrainingData
 ):
 
-    response_selector_exists = False
-    for component in pipeline:
-        # check if a response selector is part of NLU pipeline
-        if MESSAGE_RESPONSE_ATTRIBUTE in component.provides:
-            response_selector_exists = True
+    if data.regex_features and not any(
+        [component.name == "RegexFeaturizer" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data with regexes, but "
+            "your NLU pipeline does not include a RegexFeaturizer. "
+            "To featurize regexes, add a RegexFeaturizer to your pipeline."
+        )
+
+    if data.lookup_tables and not any(
+        [component.name == "CRFEntityExtractor" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data consisting of lookup tables, but "
+            "your NLU pipeline does not include a CRFEntityExtractor. "
+            "To featurize lookup tables, add a CRFEntityExtractor to your pipeline."
+        )
+
+    if data.entity_synonyms and not any(
+        [component.name == "EntitySynonymMapper" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data consisting of synonyms, but "
+            "your NLU pipeline does not include an EntitySynonymMapper. "
+            "To featurize synonyms, add a EntitySynonymMapper to your pipeline."
```

Comment:
```suggestion
            "To map synonyms, add a EntitySynonymMapper to your pipeline."
```
This one was my fault 😄